### PR TITLE
:running: Use prow/GCB for automated image builds rather than GitHub Actions

### PR DESCRIPTION
**Pull request by detiber**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
Updates the image build automation for prow/GCB instead of GitHub Actions. This PR in conjunction with https://github.com/kubernetes/test-infra/pull/14195 will complete the configuration needed.

### CODE REVIEW

**Comment by dims** [comment]
very cool to see this come together :)

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+timeout: 1200s
+steps:
+  - name: 'gcr.io/jason-heptio/gcb-docker-gcloud'
```
Comment (dims) [inline cloudbuild.yaml]: eventually this would be one of the staging repos?

Comment (detiber) [reply cloudbuild.yaml]: Yes, went ahead and marked as WIP until https://github.com/kubernetes/test-infra/pull/14219 merges and we can use that image instead.

**Comment by vincepri** [comment]
This is awesome 🎉 

LGTM, do we want to go ahead and merge this with your image or wait?

**Comment by detiber** [comment]
> LGTM, do we want to go ahead and merge this with your image or wait?

Considering the prow job to build the images is still pending, I don't see a reason to not hold until the image is ready.

**Comment by vincepri** [comment]
Can the WIP be removed?

**Pull request merged**